lib/introspect/assetdeclaration.js

```diff
@@ -55,24 +54,24 @@ class AssetDeclaration extends ClassDeclaration {
      * @return {string} the short name of the base system type
      */
     getSystemType() {
-        return 'Asset';
+        return this.modelFile.getModelManager().getSystemModelTable().get('Asset');
```

Comment:
This kind of changes worries me. What is the behavior if you create assets and try and validate against them, but the definition for `Asset` from the original Hyperledger system CTO is not loaded?

Reply:
The scenario is that we are parsing a JSON file with a given $class? The result would depend on what the root type was defined as in the system model file that created the JSON file. Is the default (nothing) was incompatible with it then validation would fail because the JSON $class would contain additional fields.

Reply:
The validation would fail in that case. It now becomes the responsibility of the calling application to define it's own base system types (see https://github.com/dselman/composer-concerto/pull/5/files#diff-b83100af8a7afb288a24ff3975497d64). We should probably have a warning if you try to add a model and there isn't a system model defined? 

The problem with adding a default system model is that a client has to remove it before they can add their own. This gets a little clumsy if you want to have multiple system models. Perhaps it's not so bad.

We definitely need more tests, though.